src/main/java/dao/DiscountDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package dao;
+
+import model.Discount;
+import model.DiscountMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+import java.util.Collection;
+
+public class DiscountDaoImpl implements DiscountDao {
+    private DataSource dataSource;
+    private JdbcTemplate jdbcTemplate;
+
+    @Override
+    public Discount getDiscount(int id) {
+
+        String SQL = "SELECT * FROM discounts WHERE id = ?";
+        return (Discount) jdbcTemplate.queryForObject(SQL, new DiscountMapper(), id);
+    }
+
+    @Override
+    public Collection<Discount> getAllDiscounts() {
+        String SQL = "SELECT * FROM discounts";
```

Comment:
Make all strings that you use for SQL queries private static final fields of  the class